### DISCUSSION TITLE
Discord hangout: 2 September 2023

### DISCUSSION THREAD

**Discussion by l3xander** [freesewing/freesewing#4888] [Support] [answered]
You are all invited to our Discord hangout.
Please leave a comment here or [on Discord](https://discord.freesewing.org/) to add things to the agenda.
This page will be updated after the call with notes.

### About
FreeSewing's Discord hangout is an informal hangout/chat/meeting that is open to everyone.
The call is organized/hosted by @tangerineshark.
The call does not have a fixed structure or agenda items. Everyone is free to suggest agenda items, ask questions, or bring ideas.

**Details**
Location: https://discord.freesewing.org (meeting room: 🔊`voice chat`)
Date: September 2nd, 2023 4pm UTC, which can be found in your local time zone [here](https://savvytime.com/converter/utc-to-belgium-antwerpen-ca-los-angeles-il-chicago-ny-new-york-city-brazil-sao-paulo-united-kingdom-london-france-paris-south-africa-johannesburg-msd-india-mumbai-hkt-japan-tokyo-australia-sydney/sep-2-2023/4pm).
Or check the Events tab in Discord!
The meetings will always be at 6pm, Antwerp time.

**Comment by tangerineshark** [comment]
Heyo, I got swamped this week but I talked to joost about the guidelines for designers from last week and can share updates!

**Comment by tangerineshark** [comment, marked as answer]
The swampening continues, but here's my notes on what we talked about in the last two calls regarding guidelines for designers and maintainers. Gonna tag the folks I remember being in this conversation and whose github handles I know, but feel free to add people I missed: @lexander2002 @BenJamesBen @bobgeorgethe3rd @SeaZeeZee @anna-puk @eriese @darigovresearch @starfetch  Thrunic Braveness @tangerineshark @joostdecock (Also in other news, Thrunic has created an awesome Lilo and Stitch onesie and Braveness' Aaron turned out fab, or maybe that was the week before):

# Guidelines for pattern designers, maintainers, and coders

**What are the different roles we're talking about here?**
- **_Designer_**: Drafts a pattern and builds it in FreeSewing, then shares it as a PR
- **_Maintainer_**: Monitors a pattern long-term, including bugs and updates
- **_Coder_**: Codes a pattern in partnership with a pattern designer or drafter
- There's probably additional categories here, like drafter (no coding), documenter, etc.

**So you're designing a pattern.... What is the definition of done?**
- **_Parametric_**: If a pattern doesn't adjust to different users' measurements, then it's probably not ready to be posted on FreeSewing.
- **_Tested on at least one person_**: Real-life testing helps identify issues, tweaks, and sewing suggestions. You can, however, test it on someone else if you're a coder who has no interest in sewing. Just, we focus on made-to-measure garments, so you should make one. To measure. (Bonus points if you test it on multiple people with different body types.)
- **_Additional suggestions_**: It would be cool if you could share at least the skeleton of your instructions for a pattern, or be willing to work with someone else who's game to write them up. Patterns don't _have_ to have instructions to get posted, but it sure makes things easier, and it's a lot more approachable for new folks to build on existing documentation than create it from scratch. 
- And don't forget, if you need help to get your pattern across the finish line, talk to the community! 

**So you're maintaining a pattern... What kind of ongoing support should you provide?**
- **_Pattern maintainers (often the designers but not always) should be willing to fix bugs that break their pattern_**: When a pattern crashes, a bug report can be submitted automatically and assigned to the pattern maintainer. Some folks also identify bugs via community discussions, or post them directly as Github issues. This means that if the pattern does something unexpected with a users' measurements, or when the software operates differently from what you intended it to do, you'll take a look and try to resolve it, either directly or by looping in other folks. It does not mean that you have to solve every request related to your pattern, which gets to the flip side of this coin.
- **_Pattern maintainers should field suggestions, fit issues, and requests for enhancements, but they are not required to implement them_**: For example, if someone wants to add a different sort of sleeve to your pattern, but you're not interested in that sleeve option, you don't have to add it. In this case, other contributors are welcome to make derivative works of a pattern. Note that patterns that inherit from other patterns may be altered as a result of changes in the original pattern. 

**Suggestions for the future**
- Could we identify patterns that lots of folks have made successfully? (It's possible the Showcase posts already take care of this.)
- We should have a clear path to connect folks that want to work on docs with developers that want documentation support.
- Add a spokesperson role for patterns that expands the pool of people that can be tagged for support in making a pattern. For example, @joostdecock is the designer and maintainer of `Paco`, but you can ask @tangerineshark for help making that pattern, too.
- If you know you don't want to maintain a pattern long-term, it would be cool to have a way to crowdsource or tag in a new maintainer.
- Would we consider a way to flag designs as experimental or needing testing? (Like a warning but friendlier!)

Note from @tangerineshark : I definitely missed some things here, so pleaeeeeaaaasesee feel free to share additional thoughts in the comments.
